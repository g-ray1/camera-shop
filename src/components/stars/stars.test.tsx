import { render } from '@testing-library/react';
import Stars from './stars';

describe('Stars component', () => {
  it('Snapshot without reviewCount', () => {
    const view = render(<Stars rating={5} />);

    expect(view).toMatchSnapshot();
  });

  it('Snapshot with reviewCount', () => {
    const view = render(<Stars rating={0} reviewCount={100}/>);

    expect(view).toMatchSnapshot();
  });
});
