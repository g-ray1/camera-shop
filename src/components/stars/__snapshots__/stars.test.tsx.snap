// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars component Snapshot with reviewCount 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rate product-card__rate"
      >
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
        <p
          class="visually-hidden"
        >
          Рейтинг:
        </p>
        <p
          class="rate__count"
        >
          <span
            class="visually-hidden"
          >
            Всего оценок:
          </span>
          100
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rate product-card__rate"
    >
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-star"
        />
      </svg>
      <p
        class="visually-hidden"
      >
        Рейтинг:
      </p>
      <p
        class="rate__count"
      >
        <span
          class="visually-hidden"
        >
          Всего оценок:
        </span>
        100
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Stars component Snapshot without reviewCount 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rate product-card__rate"
      >
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-full-star"
          />
        </svg>
        <svg
          aria-hidden="true"
          height="16"
          width="17"
        >
          <use
            xlink:href="#icon-full-star"
          />
        </svg>
        <p
          class="visually-hidden"
        >
          Рейтинг:
        </p>
        <p
          class="rate__count"
        >
          <span
            class="visually-hidden"
          >
            Всего оценок:
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rate product-card__rate"
    >
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <svg
        aria-hidden="true"
        height="16"
        width="17"
      >
        <use
          xlink:href="#icon-full-star"
        />
      </svg>
      <p
        class="visually-hidden"
      >
        Рейтинг:
      </p>
      <p
        class="rate__count"
      >
        <span
          class="visually-hidden"
        >
          Всего оценок:
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
